framework/src/problems/FEProblemBase.C

```diff
@@ -266,6 +266,11 @@ FEProblemBase::validParams()
       false,
       "Set to true to allow convergence even though the solution has been marked as 'invalid'");
 
+  params.addParam<bool>("immediately_print_invalid_solution",
+                        false,
+                        "Whether or not to report invalid solution warnings at the time the "
+                        "warning is produced instead of after the calculation");
+
```

Comment:
add a parameter group
see other large objects (fixedpointsolve, multiapp..) for example

Reply:
I feel it may be better to have these two parameters in FEProblemBase since they are only defined and used in FEProblemBase. They are not used inside SolutionInvalidity class.

Reply:
https://mooseframework.inl.gov/docs/doxygen/moose/classInputParameters.html#a248ce59e136741971a8ad8cdc95ab6b8

Reply:
Comment has been addressed. The group is named "SolutionInvalidity Output Control". Do you have suggestions on better parameter group name?

Reply:
Probably too long - I'll let @GiudGiud comment on this one